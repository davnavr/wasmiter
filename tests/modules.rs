#[test]
fn basic_module() {
    let wat = r#"(module
    (func (export "add_five") (param i32) (result i32)
        local.get 0
        i32.const 5
        i32.add))
"#;
    let wasm = wat::parse_str(wat).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn exception_handling() {
    let wasm = wat::parse_str(include_str!("modules/exception_handling.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn all_the_things() {
    let wasm = wat::parse_str(include_str!("modules/all_the_things.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}
