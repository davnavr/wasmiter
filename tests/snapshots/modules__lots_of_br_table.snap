---
source: tests/modules.rs
expression: "wasmiter::parse_module_sections(wasm.as_slice()).unwrap().display_module()"
---
(module
(type (; 0 ;) (func (param) (result)))

;; function section count = 1
(func (type 0) ;; code size = 77
  block 
    block 
      block 
        block 
          block 
            block 
              block 
                block 
                  block 
                    block 
                      block 
                        block 
                          block 
                            block 
                              block 
                                block 
                                  block 
                                    block 
                                      i32.const 0x0000000A (; 10 signed, 10 unsigned ;)
                                      br_table 2 3 4 5 6 7 8 9 10 11 12 13 14 15 14
                                      unreachable
                                      drop
                                      end
                                    end
                                  end
                                end
                              end
                            end
                          end
                        end
                      end
                    end
                  end
                end
              end
            end
          end
        end
      end
    end
)
)
